config/initializers/active_model_serializer.rb

```diff
@@ -1,3 +1,42 @@
 # frozen_string_literal: true
 
-ActiveModelSerializers.config.adapter = :json_api
+class CustomPaginationLinks < ActiveModelSerializers::Adapter::JsonApi
+  def success_document
+    res = super
+
+    if !res[:links].nil?
+      normalize_pagination_links(res[:links])
+      res[:meta] = append_pagination_meta_info(@serializer.object)
```

Comment:
You should account for the fact that the serializer might want to override the key `meta`, and this way it can't.

Reply:
Updated PR to only append the `meta` key if it is nil